frontend/public/components/secrets/create-secret.tsx

```diff
@@ -55,13 +59,17 @@ export type BasicAuthSubformState = {
   password: string;
 };
 
-const secretFormExplanation = {
-  [SecretTypeAbstraction.generic]:
-    'Key/value secrets let you inject sensitive data into your application as files or environment variables.',
-  [SecretTypeAbstraction.source]: 'Source secrets let you authenticate against a Git server.',
-  [SecretTypeAbstraction.image]:
-    'Image pull secrets let you authenticate against a private image registry.',
-  [SecretTypeAbstraction.webhook]: 'Webhook secrets let you authenticate a webhook trigger.',
+const secretFormExplanation: Record<SecretTypeAbstraction, (t: TFunction) => string> = {
```

Comment:
It's better to move code like this into a block that already has useTranslation or withTranslation rather than passing in a TFunction.

Reply:
But this is before and with this PR a utils function, not a component ?! Hmm, don't know how I could change this without refactoring much more code.

Reply:
Oh, I see, `secretFormExplanation` is being used in two components in this file. Let me see if I can find how we've handled this in other PRs. I know we've come across this before.